### PR TITLE
Remove HTML regexp

### PR DESCRIPTION
**Pull request by llimllib**
* test process_response more thoroughly
* improve process_response parsing
* improve process_response error handling

### CODE REVIEW

**Comment by harrisj** [comment]
Looks good to me. Thank you very much for saving me from my git spiral

**Comment by llimllib** [comment]
This branch doesn't pass the external integration tests:

```
1) VBMS::Requests UploadDocumentWithAssociations executes succesfully when pointed at VBMS
     Failure/Error: @client.send(request)
     VBMS::SOAPError:
       Unable to parse SOAP response
     # ./src/vbms/client.rb:155:in `rescue in process_response'
     # ./src/vbms/client.rb:152:in `process_response'
     # ./src/vbms/client.rb:94:in `send'
     # ./spec/requests/requests_spec.rb:34:in `block (4 levels) in <top (required)>'
     # ./spec/requests/requests_spec.rb:18:in `block (3 levels) in <top (required)>'

  2) VBMS::Requests FetchDocumentById executes succesfully when pointed at VBMS
     Failure/Error: webmock_soap_response(@client.endpoint_url, 'list_documents', 'listDocumentsResponse')
     RuntimeError:
       You want to finish 2 frames, but stack size is only -48
     # ./vendor/bundle/ruby/2.1.0/gems/byebug-5.0.0/lib/byebug/attacher.rb:17:in `step_out'
     # ./vendor/bundle/ruby/2.1.0/gems/byebug-5.0.0/lib/byebug/attacher.rb:17:in `attach'
     # ./vendor/bundle/ruby/2.1.0/gems/byebug-5.0.0/lib/byebug/attacher.rb:28:in `byebug'
     # ./spec/spec_helper.rb:67:in `webmock_soap_response'
     # ./spec/requests/requests_spec.rb:54:in `block (3 levels) in <top (required)>'

  3) VBMS::Requests GetDocumentTypes executes succesfully when pointed at VBMS
     Failure/Error: webmock_soap_response(@client.endpoint_url, 'get_document_types', 'getDocumentTypesResponse')
     RuntimeError:
       You want to finish 2 frames, but stack size is only -49
     # ./vendor/bundle/ruby/2.1.0/gems/byebug-5.0.0/lib/byebug/attacher.rb:17:in `step_out'
     # ./vendor/bundle/ruby/2.1.0/gems/byebug-5.0.0/lib/byebug/attacher.rb:17:in `attach'
     # ./vendor/bundle/ruby/2.1.0/gems/byebug-5.0.0/lib/byebug/attacher.rb:28:in `byebug'
     # ./spec/spec_helper.rb:67:in `webmock_soap_response'
     # ./spec/requests/requests_spec.rb:69:in `block (3 levels) in <top (required)>'
```

**Comment by llimllib** [comment]
It fails on UploadDocumentWithAssociations because UDWA returns a multipart response, not just XML. Example:

```
--uuid:a10f73c8-60e9-4985-ab2c-ac5fcd8baf2d\r\nContent-Type: application/xop+xml; charset=UTF-8; type=\"text/xml\"\r\nContent-Transfer-Encoding: binary\r\nContent-ID: <root.message@cxf.apache.org>\r\n\r\n<soap:Envelope xmlns:soap=\"http://schemas.xmlsoap.org/soap/envelope/\"><env:Header xmlns:env=\"http://schemas.xmlsoap.org/soap/envelope/\"><wsse:Security xmlns:wsse=\"http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-secext-1.0.xsd\" xmlns:wsu=\"http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-utility-1.0.xsd\"><xenc:EncryptedKey xmlns:xenc=\"http://www.w3.org/2001/04/xmlenc#\" Id=\"EK-AE8A3D115F35F18F471441467996268232\"><xenc:EncryptionMethod Algorithm=\"http://www.w3.org/2001/04/xmlenc#rsa-1_5\"/><ds:KeyInfo xmlns:ds=\"http://www.w3.org/2000/09/xmldsig#\"><wsse:SecurityTokenReference><ds:X509Data><ds:X509IssuerSerial><ds:X509IssuerName>CN=cui to VBMS,OU=VBMS,O=VA,L=Charleston,ST=SC,C=US</ds:X509IssuerName><ds:X509SerialNumber>10750083120540719457</ds:X509SerialNumber></ds:X509IssuerSerial></ds:X509Data></wsse:SecurityTokenReference></ds:KeyInfo><xenc:CipherData><xenc:CipherValue>H/Pl0rIG+CKgaDksE9HprufAFw0NXp8DQlvpyo4wBSWd5Pxmn/IqN1neLc5ncW0cfh2eOYzb7ili7nMSHB2o10Q8+qt7z5zpFgeVbj5RJCKWEDjip8Bx+yWkdroNbYpY8KLozSYacwBcKtABqbSl4CUY34z6nJ6Pt4Rlt/busVdqFvz/Be9Dx/I3qQSKNxFA9x5gZub2zPDQxyiBsUYCquH+2/tOroIxLcI25DJevt6kPyQf33wlkS3NDTl1/br93IQ7TzcXEAyRyfwbwKLG2EiVAQGScFhgQoZSdOnPBTYy+jE8tAsLzf0s7GyQNqdZkotF/TW6nHdcWElgOgcgZg==</xenc:CipherValue></xenc:CipherData><xenc:ReferenceList><xenc:DataReference URI=\"#ED-232\"/></xenc:ReferenceList></xenc:EncryptedKey><wsu:Timestamp wsu:Id=\"TS-229\"><wsu:Created>2015-09-05T15:46:36.228Z</wsu:Created><wsu:Expires>2015-09-05T15:51:36.228Z</wsu:Expires></wsu:Timestamp><ds:Signature xmlns:ds=\"http://www.w3.org/2000/09/xmldsig#\" Id=\"SIG-231\"><ds:SignedInfo><ds:CanonicalizationMethod Algorithm=\"http://www.w3.org/2001/10/xml-exc-c14n#\"><ec:InclusiveNamespaces xmlns:ec=\"http://www.w3.org/2001/10/xml-exc-c14n#\" PrefixList=\"soap\"/></ds:CanonicalizationMethod><ds:SignatureMethod Algorithm=\"http://www.w3.org/2000/09/xmldsig#rsa-sha1\"/><ds:Reference URI=\"#TS-229\"><ds:Transforms><ds:Transform Algorithm=\"http://www.w3.org/2001/10/xml-exc-c14n#\"><ec:InclusiveNamespaces xmlns:ec=\"http://www.w3.org/2001/10/xml-exc-c14n#\" PrefixList=\"wsse soap\"/></ds:Transform></ds:Transforms><ds:DigestMethod Algorithm=\"http://www.w3.org/2000/09/xmldsig#sha1\"/><ds:DigestValue>/vvgN4HTwK1rw9cLSW7+lnrPlEc=</ds:DigestValue></ds:Reference><ds:Reference URI=\"#id-230\"><ds:Transforms><ds:Transform Algorithm=\"http://www.w3.org/2001/10/xml-exc-c14n#\"><ec:InclusiveNamespaces xmlns:ec=\"http://www.w3.org/2001/10/xml-exc-c14n#\" PrefixList=\"\"/></ds:Transform></ds:Transforms><ds:DigestMethod Algorithm=\"http://www.w3.org/2000/09/xmldsig#sha1\"/><ds:DigestValue>FRP1R4Zl6kB+Hq0SJv7yiHBx1e8=</ds:DigestValue></ds:Reference></ds:SignedInfo><ds:SignatureValue>HYUABv6qa+ohKHVira5hQimMEcZl03CAIMxED5xdNhoNi8Rpjub6SjOHN5FXvUiWr4RL7aYVDkhs\n6vyyQfbddWU65QzApg1W+wNVigdG0Hm6GykSZHZHXjpcHdcQRataZAdky333YY4Awt7PrqvKT/mq\n2Ni925u+bs2Zb2SCKJ4OGDOwO5c2A294lcnkF+TsomtkEFxHfHHY6EwzedUiqmZ17Hs4A1bwcjSN\n2tOU/qA4ZyQnYQtVj59vV4nIBWcNFCVD3zM35ZXwssd0abf6YcbF+STJ6D27JzsqUsAZnmakMfoT\nLp1ecEgbGvOOUWEQ57mZm/e8nS3R9+A/1SUTmg==</ds:SignatureValue><ds:KeyInfo Id=\"KI-AE8A3D115F35F18F471441467996229230\"><wsse:SecurityTokenReference wsu:Id=\"STR-AE8A3D115F35F18F471441467996229231\"><ds:X509Data><ds:X509IssuerSerial><ds:X509IssuerName>CN=AIDE CA2,OU=CA2,O=AIDE,L=Culpeper,ST=Virginia,C=US</ds:X509IssuerName><ds:X509SerialNumber>236</ds:X509SerialNumber></ds:X509IssuerSerial></ds:X509Data></wsse:SecurityTokenReference></ds:KeyInfo></ds:Signature></wsse:Security></env:Header><soap:Body xmlns:wsu=\"http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-utility-1.0.xsd\" wsu:Id=\"id-230\"><xenc:EncryptedData xmlns:xenc=\"http://www.w3.org/2001/04/xmlenc#\" Id=\"ED-232\" Type=\"http://www.w3.org/2001/04/xmlenc#Content\"><xenc:EncryptionMethod Algorithm=\"http://www.w3.org/2001/04/xmlenc#aes128-cbc\"/><ds:KeyInfo xmlns:ds=\"http://www.w3.org/2000/09/xmldsig#\"><wsse:SecurityTokenReference xmlns:wsse=\"http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-secext-1.0.xsd\" xmlns:wsse11=\"http://docs.oasis-open.org/wss/oasis-wss-wssecurity-secext-1.1.xsd\" wsse11:TokenType=\"http://docs.oasis-open.org/wss/oasis-wss-soap-message-security-1.1#EncryptedKey\"><wsse:Reference URI=\"#EK-AE8A3D115F35F18F471441467996268232\"/></wsse:SecurityTokenReference></ds:KeyInfo><xenc:CipherData><xenc:CipherValue>nR4sD1aMBu2/m4KBRo4aG36OzhPjRutf7E5RHFI6G++EQxFbW1u+fQE05LoaoY+a/pAWvD/tJyaGx5YvT/XKnePV/VoVx6BaDf7Off2jhv1ONrXsPwDupzvHFJhpjhNkkUrGid4qJXrIWMRzmXXx5iJuR0FuT3owIzbHKmpB9CTsHWppTpJO9X+4YgFaPWkvbkUFLElqqGHZekcaqTiisrL0WFfrG+RujRkPSAYQEJqCA30NuHEv7IVdIKbf+WQhMNvOI+ekNcaZ3/cuxUpx4wTBZf1YKE/qJX//s2bbPbpPxoPiZjGq3SJck4mGPFnEfrFTAvQu1Oxi5T7G8sEuwJiJXuQtU9lrsnLQpDo7lr2goieuw8x2SpsU+1Tdo8wbzepycVn41UazOzLoWMf7r47jw/OzdVnBnvwJbCXpjqIR6ChakoATCzY+pszXsRBHywcf21PImwrLfmCK/bEZApLRdbgcmbewW9X6AEyY1dBCtO2GMbkc1abQ45Kdki3vUMR/uY5Vnt6P/vkYP7SmmTA036aCSNW44TGoYQOaoKb1MDQIc+NV6PPdiOcX3OVRkDcLBoJqA4/bfT4xWLP/MfSUZZWrXLt4LFcv7A6zDWpWjrZwQ2fycYwUSelcj7Y4moXJOTyd6cQ6t4rlpECnjCpMyqNDMxh9GxhRlagrZc2NE7+GYU9dLtRxDoVEO5JEaKySgBunTxRspIRmvFX3+788a4tVnZSXXWaSdgX4TDOmjojT3Nm+WtRK6qFGYp8SBKtsjBeADD2YI44lTJdSIsIwk9l9NGNpaK1SMzYFp2l5yrNZxfNGo+OcjKhGiIc30Wsdbz3tyqNuK0DF+cb08+9Mv6nOJ4GneS5Tk4kg86pPUB1tKBnDuCsRXzRrLy5NFir75cer5J3mr+baqNZb/QKdA52/XGvrcn34Z8WWgpjLSShR2KBKF4XYtQ81rC9rgbog994mTGGHddPK0vOnaS5ttsotkA8E98gL7Y1FsgZlBrccNPvwgCYwxFZ2CXeXboQcR5yfff3QiG77SncYK6Z88PC5kUlGdGhwrWYQXSV4UeA+egBahHfEnQ4PW+oyM4b/6W4g8izqaNfomKWEYvmhv+LYBwfCbt7KYNdXXHFhwDSxBAuXqUJyrgFApWZmj+QsMMhVzYqa+gf3CgvDGBb/pQQ8SqohCN6N2gsD6idwtKUeB7vahLVnagkvL5XPRSz4TYkQqnu2vTCHuLojr1B+WyQxYU+F8xJSvzbIP8Jh+HyvoXSVvoasdMwWZfhXdLcRMHX/rB3DqAMiCf1vwkHtp4KTjMk9ds7hNb8JJq1rRG7HYBwxODnFgXT7CSRWg2naQBdfZ7EkGSH1j1xPSP24nlTSOBY0HuIqcceJv3cNKa+/U+9VzVZeVnQ2xpP25R39Cimp3PWl/XYWe8/gYuxn0DMiMoYvQnq4QUFu5/KUdj3Bj560maEb/BwlfTTN21YtPv3Nzf4B+OrISk0usm1L8oUQLze3e/5cjlfDHKi02Rjd2G9s6Lx8kwKLFVvJhbNh4jyqRjhW78LYJpj4YphOoj+EjIEGwYQfLR15M0fYiJtX6j+WBfGzxTzyfPD5B87yWlrLZtspAk4/U0k2mAKk2MHtp8zcIMlRGjMIRZZvDUXctoLQ5JmbnMkHYbAC5U3zQnYsExhxpvj1h0D4IHD9BDemFucdE4q3SQJ8NlHAWW379nQo6sbTjXv9oxjVADXPAvOeKjvxnLAvbPiquPAiMI+3VHt1h1K2HpMk2AKEkpfp8+/Ehltw+EtVmIOl9bX/tY2ctyOXF4ZxY4zUuuuSceR7Q9APrSvmx26zJNC409KLllBhzNUQstqbS3grCSn53Ge0zt9Ap6UbHDVlTlBhptbKRwCAiZ+QFNtEpR6dlEF0LPlbbiGb/bid3W6/vVwJLgO1aviQEJBF2Q7Pn3mi0K/smdY1c7XLFp7apnMYlJN1o5wwvN7ZNw7pUHUiClhgfUYu8ENoHg69d8hHSfMOBttlSru5/ACEV5HLvWEknJtMCe2jW00lBdhoABGJWWOUw3mfRE4a4EqyvP74tnyEzYNl9PjPPoQyaQJLbK0m1stXnMkujxADZzKkCOKT</xenc:CipherValue></xenc:CipherData></xenc:EncryptedData></soap:Body></soap:Envelope>\r\n--uuid:a10f73c8-60e9-4985-ab2c-ac5fcd8baf2d--
```

**Comment by llimllib** [comment]
(now I remember why I used a regex!)

**Comment by harrisj** [comment]
Oops! Clearly my webmocks are bad. Will fix.

**Comment by llimllib** [comment]
If there's a widely available ruby library for parsing MIME messages, I haven't found it

**Comment by llimllib** [comment]
OK, give #52 a look; I think it handles these responses properly

File: src/vbms/client.rb [hunk 1]
```diff
@@ -141,23 +141,47 @@ def build_request(body, headers)
...
+      # the envelope should be the root node of the document
+      soap = doc.at_xpath("/soapenv:Envelope", VBMS::XML_NAMESPACES)
+      fail SOAPError.new("No SOAP envelope found in response", response.body) if
+        soap.nil?
+
+      fail SOAPError.new("SOAP Fault returned", response.body) if
+        soap.at_xpath('//soapenv:Fault', VBMS::XML_NAMESPACES)
+    end
   end
 end
```
Comment (amoose) [inline src/vbms/client.rb]: The rubocop:disable comments were included with the initial implementation of rubocop, however, let's try not to add exclusions going forward if at all possible. I DRYd out the process_response method and removed the rubocop exception comments.

**Comment by amoose** [comment]
This is GTG @llimllib if you're :ok_hand: with the last commit (removing rubocop:disable).

**Comment by harrisj** [comment]
I am pro this being merged if you all are.

**Pull request closed without merging**